app/models/deploy_metrics.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Used to collect metrics of cycle time(elapsed time needed to deliver a project to production)
+# Check out https://about.gitlab.com/2019/03/04/reduce-cycle-time/ for more details on cycle time
+# The main objective for this is to gain more insight on the efficiency of the end-to-end project development process
+# There are 2 key metrics:
+# 1. PR - Production: this will show the time needed for a pull request to get deployed to production environment
+# 2. Staging - Production: this will show the time needed to deliver from staging(non-production environment)
+#                          to production environment.
+class DeployMetrics
+  NOT_APPLICABLE = 0
+
+  def initialize(deploy)
+    @deploy = deploy
+  end
+
+  def cycle_time
+    return {} unless @deploy&.production && @deploy&.status == "succeeded"
+
+    {pr_production: pr_production, staging_production: staging_production}
```

Comment:
atm there are 2 ways of excluding a metric, either by not setting the key or by setting it to NOT_APPLICABLE
ideally there should be only 1

so how about:
```ruby
times = {}
if time = pr_production
  times[:pr_production] = time
end
if time = staging_production
  times[:staging_production] = time
end
times
```

and inside the methods just do `return unless foo` so the methods return `nil`

Reply:
okay sure